http-server/src/main/java/io/airlift/http/server/HttpRequestEvent.java

```diff
@@ -42,8 +43,8 @@ public static HttpRequestEvent createHttpRequestEvent(Request request, Response
             user = principal.getName();
         }
 
-        String token = null;
-        if (traceTokenManager != null) {
+        String token = request.getHeader(TRACETOKEN_HEADER);
+        if (token == null && traceTokenManager != null) {
```

Comment:
I would expect that the header token would be in the token manager already from the `TraceTokenFilter`

Reply:
If requests are processed synchronously then yes the token will be available in thread locals, but for async processing threads other than the Jetty worker threads will go through this code path and the token won't be available in the thread locals.